src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.spec.ts

```diff
@@ -0,0 +1,153 @@
+import { PipelinesWidgetComponent } from './pipelines-widget.component';
+
+import { LocationStrategy } from '@angular/common';
+import { Component, NO_ERRORS_SCHEMA } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { HttpModule } from '@angular/http';
+import { By } from '@angular/platform-browser';
+import { ActivatedRoute, Router, RouterModule } from '@angular/router';
+import { Broadcaster } from 'ngx-base';
+import { BsModalService } from 'ngx-bootstrap/modal';
+import { Context, Contexts, Space } from 'ngx-fabric8-wit';
+import { AuthenticationService, User, UserService } from 'ngx-login-client';
+import { ConnectableObservable, Subscription } from 'rxjs';
+import { Observable, Subject } from 'rxjs';
+import { createMock } from 'testing/mock';
+import {
+  initContext,
+  TestContext
+} from 'testing/test-context';
+import { BuildConfig } from '../../../a-runtime-console/index';
+import { PipelinesService } from '../../space/create/pipelines/services/pipelines.service';
+
+@Component({
+  template: '<fabric8-pipelines-widget></fabric8-pipelines-widget>'
+})
+class HostComponent { }
+
+describe('PipelinesWidgetComponent', () => {
+  type TestingContext = TestContext<PipelinesWidgetComponent, HostComponent>;
+
+  let modalService: jasmine.SpyObj<BsModalService>;
+  let broadcaster: jasmine.SpyObj<Broadcaster>;
+  let pipelinesService: jasmine.SpyObj<PipelinesService>;
+  let authentication: jasmine.SpyObj<AuthenticationService>;
+  let ctxSubj: Subject<Context> = new Subject();
+
+  let mockRouter;
+  let mockActivatedRoute;
+  let mockLocationStrategy;
+
+  let fakeUserObs: Subject<User> = new Subject<User>();
+
+  let mockRouterEvent: any = {
+    'id': 1,
+    'url': 'mock-url'
+  };
```

Comment:
Can this be moved into the factory function? It doesn't appear that you reference it anywhere below in the test cases, only in the Router factory.